edx_toggles/toggles/internal/waffle.py

```diff
@@ -56,13 +56,39 @@ def is_enabled(self, switch_name):
         """
         Returns and caches whether the given waffle switch is enabled.
         """
+        value = self.get_request_cache(switch_name)
         namespaced_switch_name = self._namespaced_name(switch_name)
-        value = self._cached_switches.get(namespaced_switch_name)
         if value is None:
             value = switch_is_active(namespaced_switch_name)
-            self._cached_switches[namespaced_switch_name] = value
+            self.set_request_cache(namespaced_switch_name, value)
         return value
 
+    def get_request_cache(self, namespaced_switch_name, default=None):
+        """
+        API for accessing the request cache. In general, users should avoid accessing the namespace cache.
+        """
+        return self._cached_switches.get(namespaced_switch_name, default)
+
+    def get_request_cache_with_short_name(self, switch_name, default=None):
```

Comment:
I like that this doesn't need to be used internally, and will be clearly temporary wherever it is used in edx-platform.